src/main/kotlin/com/cjbooms/fabrikt/generators/controller/MicronautControllerInterfaceGenerator.kt

```diff
@@ -149,9 +166,40 @@ class MicronautControllerInterfaceGenerator(
             )
         }
 
+
+        var securityOption = op.getSecurityRequirements().securitySupport(op.hasSecurityRequirements())
+
+        if(securityOption == SecuritySupport.NO_SECURITY) {
+            securityOption = globalSecurity
+        }
+
+        val securityRule = setSecurityRule(securityOption)
+
+        if(securityRule != "") {
+            this.addAnnotation(
+                AnnotationSpec
+                    .builder(MicronautImports.SECURED)
+                    .addMember(
+                        securityRule
+                    )
+                    .build()
+            )
+        }
+
         return this
     }
 
+    private fun setSecurityRule(
+        securityOption: SecuritySupport?,
+    ): String {
+        return when (securityOption) {
+            SecuritySupport.AUTHENTICATION_REQUIRED -> SECURITY_RULE_IS_AUTHENTICATED
+            SecuritySupport.AUTHENTICATION_PROHIBITED -> SECURITY_RULE_IS_ANONYMOUS
+            SecuritySupport.AUTHENTICATION_OPTIONAL -> SECURITY_RULE_IS_AUTHENTICATED + ", " + SECURITY_RULE_IS_ANONYMOUS
```

Comment:
**suggestion:** `"$SECURITY_RULE_IS_AUTHENTICATED, $SECURITY_RULE_IS_ANONYMOUS"`

Reply:
Much better this way!